xmppserver/src/main/java/org/jivesoftware/openfire/net/DNSUtil.java

```diff
@@ -366,128 +372,66 @@ public static boolean isNameCoveredByPattern( String name, String pattern )
 
     /**
      * Encapsulates a hostname and port.
+     *
+     * @deprecated Replaced by {@link SrvRecord}
      */
+    @Deprecated(since = "4.10.0", forRemoval = true) // Remove in or after Openfire 4.11.0
     public static class HostAddress implements Serializable {
 
-        private final String host;
-        private final int port;
-        private final boolean directTLS;
+        private final SrvRecord delegate;
 
         public HostAddress(String host, int port, boolean directTLS) {
-            // Host entries in DNS should end with a ".".
-            if (host.endsWith(".")) {
-                this.host = host.substring(0, host.length()-1);
-            }
-            else {
-                this.host = host;
-            }
-            this.port = port;
-            this.directTLS = directTLS;
+            delegate = new SrvRecord(host, port, directTLS);
         }
 
-        /**
-         * Returns the hostname.
-         *
-         * @return the hostname.
-         */
         public String getHost() {
-            return host;
+            return delegate.getHostname();
         }
 
-        /**
-         * Returns the port.
-         *
-         * @return the port.
-         */
         public int getPort() {
-            return port;
+            return delegate.getPort();
         }
 
-        public boolean isDirectTLS()
-        {
-            return directTLS;
+        public boolean isDirectTLS() {
+            return delegate.isDirectTLS();
         }
 
         @Override
         public String toString() {
-            return host + ":" + port;
+            return delegate.getHostname() + ":" + delegate.getHostname();
         }
     }
 
+    /**
+     * @deprecated Replaced by {@link SrvRecord#prioritize(SrvRecord[])}
+     */
+    @Deprecated(since = "4.10.0", forRemoval = true) // Remove in or after Openfire 4.11.0
     public static List<Set<WeightedHostAddress>> prioritize(WeightedHostAddress[] records) {
         return prioritize(Arrays.asList(records));
     }
 
+    /**
+     * @deprecated Replaced by {@link SrvRecord#prioritize(Collection)}
+     */
+    @Deprecated(since = "4.10.0", forRemoval = true) // Remove in or after Openfire 4.11.0
```

Comment:
You changed the return value of this method in this set of commits already. Existing code is already not backward compatible - perhaps just remove it?

Reply:
If possible, it would be good to retain that (to prevent issues with plugins or third parties), and keep it in as a deprecated method.

I believe I've resolved this by restoring the original method signature now.

Reply:
Sorry, got confused between the commits :)

Reply:
No, you correctly flagged it. I changed the code earlier today. Thanks!